src/google/protobuf/compiler/csharp/csharp_message.cc

```diff
@@ -466,10 +466,24 @@ void MessageGenerator::GenerateMergingMethods(io::Printer* printer) {
     "  switch(tag) {\n");
   printer->Indent();
   printer->Indent();
-  printer->Print(
-    "default:\n"
-    "  input.SkipLastField();\n" // We're not storing the data, but we still need to consume it.
-    "  break;\n");
+  // Some descriptor messages need to store unknown fields so that options can be parsed later.
+  // We don't decide which messages need that within protoc - we use a partial method, and
```

Comment:
I think its better to decide which messages need to store unknown fields in protoc. It's a waste of code for descriptors other than options. Though only for descriptor.proto file, but worth to do so to make the code more clear.